src/lib/dot-matrix/Chart.tsx

```diff
@@ -0,0 +1,62 @@
+import React from 'react';
+import { v4 } from 'uuid';
+import { getNumberOfDots, getContainerWidth } from './utils/utils';
+import {
+  Elements,
+  DEFAULT_COLUMNS,
+  DEFAULT_ROWS
+} from "./constants";
+import { ChartProps, DataPointType } from './types';
+import classes from './styles.module.scss';
+
+const Chart = (props: ChartProps) : JSX.Element => {
+  const {
+    dimensions = {},
+    getStyles,
+    data,
+    overlappingValues,
+    total
+  } = props;
+  const {
+    rows = DEFAULT_ROWS,
+    columns = DEFAULT_COLUMNS
+  } = dimensions;
+  return (
+    <div
+      className={classes.dotsContainer}
+      style={{
+        width: `${getContainerWidth(columns)}rem`,
+        ...getStyles(Elements.DotsContainer)
+      }}
+    >
+      {data?.map((eachPoint: DataPointType, rowIndex: number) => (
+        <React.Fragment key={v4()}>
+          {eachPoint && Array.apply(null, Array(getNumberOfDots(eachPoint, rows, columns, total))).map((item: null, columnIndex: number) => (
+            <div id="dot-matrix-dots" key={v4()}>
+              {(columnIndex === 0 && rowIndex > 0 && overlappingValues[rowIndex - 1] < 1 && overlappingValues[rowIndex - 1] !== 0 && (
```

Comment:
Can we move this logic an `hasOverlapping(overlappingValues, rowIndex, columnIndex)`?